python/ray/tune/trainable/trainable.py

```diff
@@ -178,12 +179,36 @@ def __init__(
         self._monitor = UtilMonitor(start=log_sys_usage)
 
         self.remote_checkpoint_dir = remote_checkpoint_dir
-        self.custom_syncer = custom_syncer
-        self.sync_timeout = sync_timeout
+        # If no sync_config is provided, but we saving to a remote_checkpoint_dir,
+        # then provide a default syncer. `upload_dir` here is just a dummy directory
+        # that tells the SyncConfig to create a default syncer.
+        self.sync_config = sync_config or SyncConfig(
+            upload_dir=self.remote_checkpoint_dir, syncer="auto"
+        )
+
+        # TODO(ml-team): `custom_syncer` and `syncer` are deprecated. Remove in 2.6.
+        warning_message = (
+            "Specifying `custom_syncer` and `sync_timeout` as arguments in the "
+            "Trainable constructor is deprecated and will be removed in version 2.6. "
+            "Pass in a `tune.SyncConfig` object through the `sync_config` "
+            "argument instead."
+        )
+        if custom_syncer:
+            warnings.warn(warning_message, DeprecationWarning)
+            self.sync_config.syncer = custom_syncer
+        else:
+            # Resolves syncer="auto" to an actual syncer if needed
+            self.sync_config.syncer = get_node_to_storage_syncer(self.sync_config)
```

Comment:
Just to confirm: It's okay to mess with the sync config here, since it's not the user's copy anymore.

Reply:
Yeah this is fine (probably not ideal but ok for now)